src/Catty/FormulaEditor/ViewController/FormulaEditorViewController.m

```diff
@@ -1182,7 +1182,8 @@ - (void)deleteList: (id<UserDataProtocol>)userList atRow:(NSInteger)row isProjec
 - (BOOL)isVariableUsed:(UserVariable*)variable
 {
     if([self.object.project.userData isProjectVariable:variable]) {
-        for(SpriteObject *spriteObject in self.object.project.objectList) {
+        NSArray <SpriteObject*> *allObjects = [[NSMutableArray alloc] initWithArray: self.object.project.allObjects];
```

Comment:
Is there a reason why you initialized a new mutable array in this and the two methods in the changes below? I would recommend to use `allObjects` directly (there is no need for a mutable array).